package/ios/Core/CameraSession.swift

```diff
@@ -266,7 +266,10 @@ class CameraSession: NSObject, AVCaptureVideoDataOutputSampleBufferDelegate, AVC
   public final func captureOutput(_ captureOutput: AVCaptureOutput, didOutput sampleBuffer: CMSampleBuffer, from connection: AVCaptureConnection) {
     switch captureOutput {
     case is AVCaptureVideoDataOutput:
-      onVideoFrame(sampleBuffer: sampleBuffer, orientation: connection.orientation)
+      guard let device = videoDeviceInput?.device else {
+        return
+      }
+      onVideoFrame(sampleBuffer: sampleBuffer, orientation: connection.getOrientation(device: device))
```

Comment:
Same story here - I think `connection.orientation` should not need a device. Either we do the rotation here (`let relativeOrientation = connection.orientation.rotatedBy(...)`), or we just pass in `connection.orientation` directly.

I don't really get this part either - why can we not just pass `connection.orientation` here? Isn't that what you want in a Frame Processor? cc @thomas-coldwell

Reply:
Because orientation is not treated in the same way on iOS 17 as it was before.
On one of them, you have to take sensor orientation into account.

I don't really know how to solve this problem easily while still using device.sensorOrientation (because I could just add a +90 when using videoRotationAngleForHorizonLevelPreview/videoRotationAngleForHorizonLevelCapture, but that wouldn't be very clean).

Reply:
The Modern Rotation thingy is what is actually the truth - if it says `landscapeLeft` when you hold your phone in portrait, then we apply landscape left. Internally it knows what the Camera's sensor orientation is.

In the Legacy Rotation thingy we should mimic the Modern one, so also take sensor orientation into account. I think this is the place where we should handle that then.
Same for mirroring (`if device.position == .front`), flip it 180.